src/librustc_target/spec/mod.rs

```diff
@@ -878,6 +878,10 @@ pub struct TargetOptions {
 
     /// Additional arguments to pass to LLVM, similar to the `-C llvm-args` codegen option.
     pub llvm_args: Vec<String>,
+
+    /// Whether to use legacy .ctors initialization hooks rather than .init_array. Defaults
+    /// to false (uses .init_array).
+    pub use_ctors_section: bool,
```

Comment:
Maybe NetBSD should still override this to true? Or dynamically detect that for `< 9` [like clang does](https://github.com/llvm/llvm-project/blob/5a0d8c31a32711dd1101ebcd0a563dec13efdfb9/clang/lib/Driver/ToolChains/NetBSD.cpp#L501-L502).

Reply:
I've made NetBSD use `.ctors` by default. Dynamic detection seems awkward since it will inhibit cross compilation. If someone really wants `.init_array` on NetBSD 9+, we should either have a new target for different features, or they can use the -Z flag i added to manually flip behavior.